### PR TITLE
Implement continuation token size limit

### PR DESCRIPTION
**Pull request by bambriz**
# Description

This PR adds continuation token size limit support to the python sdk. The new option for item queries accepts a non-negative integer value and is represented in terms of KB, ie 2 is 2kb or 2048 Bytes. This PR fixes [issue #27625](https://github.com/Azure/azure-sdk-for-python/issues/27625)

sample:
 `query_iterable = container.query_items(query="SELECT * FROM c where c.some_field = 'some_value'", max_item_count=max_count, partition_key=Some_Partition_Key, response_continuation_token_limit_in_kb=8)`
 
This query will limit the continuation token size to 8KB. 

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .gitignore
Comment (simorenoh) [inline .gitignore:1]: please re-add the empty line, we generally want our files to have one at the end

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py [hunk 1]
```diff
@@ -320,6 +320,8 @@ def query_items(
         :keyword dict[str, str] initial_headers: Initial headers to be sent as part of the request.
         :keyword response_hook: A callable invoked with the response metadata.
         :paramtype response_hook: Callable[[Dict[str, str], AsyncItemPaged[Dict[str, Any]]], None]
+        :keyword int response_Continuation_Token_Limit_In_Kb: is used to limit the length of continuation token
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py:323]: let's use pythonic code here and have only lower case with underscores as opposed to camel case

**Review by simorenoh — CHANGES_REQUESTED** [review]
Couple small comments here and there, implementation looks good overall though.

Also missing an update to the Changelog and the addition of samples to show users how the option can be used. Please add those and ask for another review, thanks!

File: .gitignore
Comment (bambriz) [reply .gitignore:1]: The empty line is there. I just temporarily added the log files but I forgot to remove them before I made a commit so I had to make another to remove them.

**Comment by simorenoh** [comment]
Seems like the pipelines are complaining but otherwise LGTM! Let's try to get other reviews so we can release this after the next stable release.

File: sdk/cosmos/azure-cosmos/azure/cosmos/container.py [hunk 2]
```diff
@@ -354,6 +354,8 @@ def query_items(
         :keyword str session_token: Token for use with Session consistency.
         :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
         :keyword Callable response_hook: A callable invoked with the response metadata.
+        :keyword int response_continuation_token_limit_in_kb: is used to limit the length of continuation token
+            in the query response. Valid values are >= 0.
```
[the remark this answers is not in the document]
Comment (bambriz) [reply sdk/cosmos/azure-cosmos/azure/cosmos/container.py:358]: For the 0 value. 0 is a valid value for the backend, while negative values raise an error. The behaviour for 0 is basically the same as not passing in a value in the first place. In the other SDKs it is mentioned that valid values are >= 0, but saying non negative values is also valid.

File: sdk/cosmos/azure-cosmos/azure/cosmos/container.py [hunk 3]
```diff
@@ -354,6 +354,8 @@ def query_items(
         :keyword str session_token: Token for use with Session consistency.
         :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
         :keyword Callable response_hook: A callable invoked with the response metadata.
+        :keyword int response_continuation_token_limit_in_kb: **provisional keyword** The size limit in kb of the response continuation token
+            in the query response. Valid values are non-negative integers. A value of 0 is the same as not passing a value (default no limit).
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/azure/cosmos/container.py:358]:
```suggestion
            in the query response. Valid values are positive integers. A value of 0 is the same as not passing a value (default no limit).
```

File: sdk/cosmos/azure-cosmos/samples/document_management.py [hunk 4]
```diff
@@ -191,6 +191,23 @@ def delete_all_items_by_partition_key(db, partitionkey):
     for doc in item_list:
         print('Item Id: {0}; Partition Key: {1}'.format(doc.get('id'), doc.get("company")))
 
+
+def query_items_with_continuation_token_size_limit(container, doc_id):
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/samples/document_management.py:195]: thanks for adding the sample :)

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
There are no tests for this feature. 
@bambriz - for every feature we add, we should aim to add tests to make sure we test it and catch future regressions.

File: sdk/cosmos/azure-cosmos/azure/cosmos/container.py (hunk 3 above)
Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/azure/cosmos/container.py:358]: same here as above

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 #### Features Added
 * Added conditional patching for Patch operations. See [PR 30455](https://github.com/Azure/azure-sdk-for-python/pull/30455).
+* Added **provisional** ability to limit Continuation Token size. See [PR 30600](https://github.com/Azure/azure-sdk-for-python/pull/30600)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:7]:
```suggestion
* Added **provisional** ability to limit Continuation Token size when querying for items. See [PR 30600](https://github.com/Azure/azure-sdk-for-python/pull/30600)
```

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**